stdlib/src/collections/list.mojo

```diff
@@ -124,6 +124,18 @@ struct List[T: CollectionElement](CollectionElement, Sized):
         for value in values:
             self.append(value[])
 
+    fn __init__(inout self: Self, ptr: AnyPointer[T], size: Int, capacity: Int):
```

Comment:
```suggestion
    fn __init__(inout self: Self, data: AnyPointer[T], size: Int, capacity: Int):
```
To get the pointer in a list, we do `my_list.data`, I don't think we need to introduce a new name that users will have to remember

Reply:
Makes sense. Pushed changes